CMakeLists.txt

```diff
@@ -77,15 +78,21 @@ if (CMAKE_VERSION VERSION_GREATER 3.7.8)
     # add natvis file to the library so it will automatically be loaded into Visual Studio
     if(VS_ADD_NATIVE_VISUALIZERS)
         target_sources(GSL INTERFACE
-            ${CMAKE_CURRENT_SOURCE_DIR}/GSL.natvis
+            $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/GSL.natvis>
         )
     endif()
 endif()
 
+install(TARGETS GSL EXPORT Microsoft.GSLConfig
+  PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
```

Comment:
Because you're using `$<INSTALL_INTERFACE:...>` above and `install(DIRECTORY include/gsl ...)` below, you shouldn't use `PUBLIC_HEADER` here (it does nothing).